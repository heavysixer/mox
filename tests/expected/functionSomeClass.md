# PackageName - v0.0.1

## `Methods`

* [`SomeClass`][0]
* [`someClassMethod`][1]
* [`SomeClass2`][2]
* [`someClassMethod2`][3]

## `Declarations`

* [`aProperty`][4]

## `Methods`

### `SomeClass`

[\#][0] [Ⓣ][5]

Example:

    SomeClass.someClassMethod(x);
    

#### Arguments

1. `param` _(String)_ - a parameter

---

### `someClassMethod`

[\#][1] [Ⓣ][5]

Example:

    SomeClass.someClassMethod(x);
    

#### Arguments

1. `methodParam` _(Function | String)_ - some method param

#### Returns  
  
_(Function | String)_ - A return value

---

### `SomeClass2`

[\#][2] [Ⓣ][5]

Example:

    SomeClass2.someClassMethod(x);
    

#### Arguments

1. `param` _(String)_ - a parameter

---

### `someClassMethod2`

[\#][3] [Ⓣ][5]

Example:

    SomeClass2.someClassMethod(x);
    

#### Arguments

1. `methodParam` _(Function)_ - some method param
2. `methodParam2` _(Function)_ - some method param

#### Returns  
  
_(String)_ - A return value

---

## `Declarations`

### aProperty

[\#][4] [Ⓣ][6]

Some property description

---

### aPropertyWithType

[\#][7] [Ⓣ][6]

#### Types

  * _String_

Some property with type description

---



[0]: #someclass
[1]: #someclassmethod
[2]: #someclass2
[3]: #someclassmethod2
[4]: #aproperty
[5]: #methods
[6]: #declarations
[7]: #apropertywithtype