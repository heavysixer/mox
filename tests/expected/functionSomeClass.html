<!DOCTYPE html>
<html>
  <body>
    <h2><code>Methods</code></h2>
    <ul>
      <li><a href="#someclass"><code>SomeClass</code></a></li>
      <li><a href="#someclassmethod"><code>someClassMethod</code></a></li>
      <li><a href="#someclass2"><code>SomeClass2</code></a></li>
      <li><a href="#someclassmethod2"><code>someClassMethod2</code></a></li>
    </ul>
    <h2><code>Declarations</code></h2>
    <ul>
      <li><a href="#aproperty"><code>aProperty</code></a></li>
    </ul>
    <h2><code>Methods</code></h2>
    <h3><code>SomeClass</code></h3>
    <p>
      <!--Current comment--><a href="#someclass">#</a><span> </span>
      <!--file table of  contents--><a href="#methods"> &#x24C9;</a>
    </p>
    <p><p>Example:</p>

<div class="highlight"><pre lang="javascript">SomeClass.someClassMethod(x);
</pre></div></p>
    <div id="comment" class="params">
      <h4>Arguments</h4>
      <ol>
        <li><code>param</code><span> </span><em><span> (</span><span>String</span><span>)</span></em><span> - </span><span>a parameter</span></li>
      </ol>
    </div>
    <div id="comment" class="return">
    </div>
    <hr>
    <h3><code>someClassMethod</code></h3>
    <p>
      <!--Current comment--><a href="#someclassmethod">#</a><span> </span>
      <!--file table of  contents--><a href="#methods"> &#x24C9;</a>
    </p>
    <p><p>Example:</p>

<div class="highlight"><pre lang="javascript">SomeClass.someClassMethod(x);
</pre></div></p>
    <div id="comment" class="params">
      <h4>Arguments</h4>
      <ol>
        <li><code>methodParam</code><span> </span><em><span> (</span><span>Function</span><span> | </span><span>String</span><span>)</span></em><span> - </span><span>some method param</span></li>
      </ol>
    </div>
    <div id="comment" class="return">
      <h4>Returns</h4><br/>
      <br/><em><span> (</span><span>Function</span><span> | </span><span>String</span><span>)</span></em><span> - </span><span>A return value</span>
    </div>
    <hr>
    <h3><code>SomeClass2</code></h3>
    <p>
      <!--Current comment--><a href="#someclass2">#</a><span> </span>
      <!--file table of  contents--><a href="#methods"> &#x24C9;</a>
    </p>
    <p><p>Example:</p>

<div class="highlight"><pre lang="javascript">SomeClass2.someClassMethod(x);
</pre></div></p>
    <div id="comment" class="params">
      <h4>Arguments</h4>
      <ol>
        <li><code>param</code><span> </span><em><span> (</span><span>String</span><span>)</span></em><span> - </span><span>a parameter</span></li>
      </ol>
    </div>
    <div id="comment" class="return">
    </div>
    <hr>
    <h3><code>someClassMethod2</code></h3>
    <p>
      <!--Current comment--><a href="#someclassmethod2">#</a><span> </span>
      <!--file table of  contents--><a href="#methods"> &#x24C9;</a>
    </p>
    <p><p>Example:</p>

<div class="highlight"><pre lang="javascript">SomeClass2.someClassMethod(x);
</pre></div></p>
    <div id="comment" class="params">
      <h4>Arguments</h4>
      <ol>
        <li><code>methodParam</code><span> </span><em><span> (</span><span>Function</span><span>)</span></em><span> - </span><span>some method param</span></li>
        <li><code>methodParam2</code><span> </span><em><span> (</span><span>Function</span><span>)</span></em><span> - </span><span>some method param</span></li>
      </ol>
    </div>
    <div id="comment" class="return">
      <h4>Returns</h4><br/>
      <br/><em><span> (</span><span>String</span><span>)</span></em><span> - </span><span>A return value</span>
    </div>
    <hr>
    <h2><code>Declarations</code></h2>
    <h3>aProperty</h3>
    <p>
      <!--Current comment--><a href="#aproperty">#</a><span> </span>
      <!--file table of  contents--><a href="#declarations"> &#x24C9;</a>
    </p>
    <p><span><p>Some property description</p></span>
    </p>
    <hr>
    <h3>aPropertyWithType</h3>
    <p>
      <!--Current comment--><a href="#apropertywithtype">#</a><span> </span>
      <!--file table of  contents--><a href="#declarations"> &#x24C9;</a>
    </p>
    <p>
      <h4>Types</h4>
      <ul></ul>
      <li><em>String</em></li><span><p>Some property with type description</p></span>
    </p>
    <hr>
  </body>
</html>